wdl-analysis/src/graph.rs

```diff
@@ -629,3 +629,40 @@ impl DocumentGraph {
             .retain(|(from, to)| !collected.contains(from) && !collected.contains(to));
     }
 }
+
+/// An extension trait for [`DocumentGraph`].
+pub trait DocumentGraphExt {
+    /// Gets all nodes that have a dependency on the given node.
+    fn dependents(
+        &self,
+        index: petgraph::graph::NodeIndex,
+    ) -> Box<dyn Iterator<Item = petgraph::graph::NodeIndex> + '_>;
+}
+
+impl DocumentGraphExt for DocumentGraph {
+    fn dependents(
+        &self,
+        index: petgraph::graph::NodeIndex,
+    ) -> Box<dyn Iterator<Item = petgraph::graph::NodeIndex> + '_> {
```

Comment:
Assuming we can make this an inherent (rather than trait) method, we should also be able to simplify this return type:
```suggestion
    ) -> impl Iterator<Item = NodeIndex> {
```